src/openlcb/SimpleStack.cxx

```diff
@@ -258,6 +258,13 @@ int SimpleStackBase::create_config_file_if_needed(const InternalConfigData &cfg,
     {
         extend = true;
     }
+    // Handle the case where the file exists but is zero bytes. This was
+    // observed on the esp32 with SD storage which does not automatically
+    // flush to disk on write.
+    if (statbuf.st_size == 0)
```

Comment:
try to compare st_size against cfg.version().end_offset() instead.

Reply:
```
[LCC] Corrupt configuration file detected, /sdcard/LCC/config is too small: 0 bytes, expected: 489 bytes
/sdcard/LCC/config is too short (0 vs 130), forcing reset.
```
The first line is from the CS code to detect corruptions, the second line is a new LOG (VERBOSE) inside this new condition check. It appears to work correctly, I'll update the PR for this adjustment.